### DISCUSSION TITLE
[V3] Validation onUpdate being ignored

### DISCUSSION THREAD

**Discussion by WheelieDev** [livewire/livewire#6163] [1. Help]
I hope I'm not just being incredibly silly and would love this to be an easy fix, but the new onUpdate functionality with validation seems to be being ignored for some reason.

My Code 

**Form**

```php
<?php

namespace App\Livewire\Forms\Auth;

use Livewire\Attributes\Rule;
use Livewire\Form;

class loginForm extends Form
{
    #[Rule('required',message:'Please enter your email address.',onUpdate: true)]
    #[Rule('email',message:'You must provide a valid email address to login.',onUpdate: true)]
    public string $email;

    #[Rule('required',message: 'Please enter your password.')]
    public string $password;


    public function login()
    {
        if (auth()->attempt(['email' => $this->email, 'password' => $this->password])) {
            return redirect()->route('dashboard');
        }
        $this->addError('email', 'Your credentials were not recognised. Please try again.');
    }
}
```
**Component - PHP** 
```PHP
<?php

namespace App\Livewire\Auth;

use App\Livewire\Forms\Auth\loginForm;
use App\View\Components\layouts\AuthLayout;
use Livewire\Component;

class Login extends Component
{
    public loginForm $form;
    public function render()
    {
        return view('livewire.auth.login')
            ->layout(AuthLayout::class, ['headerTitle' => 'Welcome to SupportManager',
                'headerSubtitle' => 'Please login to your account',
                'pageTitle' => 'Login']);

    }
    public function login()
    {
        $this->form->login();
    }
}

```

**Component - Blade**

```blade
<div>
    <form wire:submit="login">
    <x-text.text-input name='form.email' label='Email' type='text' placeholder='Email' wire:model="form.email" />
    <x-text.text-input name='form.password' label='Password' type='password' placeholder='Password' wire:model="form.password" />
    <div class='flex items-center justify-between'>
        <x-buttons.button label='Login' disabled="{{false}}" type="submit"></x-buttons.button>
        <x-buttons.text-link label='Forgot Password?' wire:click="forgotPassword" />
    </div>
    </form>
</div>

```

Video

https://github.com/livewire/livewire/assets/5488110/6b2f14c7-3321-4410-b9aa-a41b01e99ead

I would expect the required validation to kick in before running the logic as is happening here; any ideas would be greatly appreciated

**Comment by DanielSpravtsev** [comment]
Add null type to properties

```
public ?string $email;

or 

public string $email = '';

```

**Comment by WheelieDev** [comment]
Thank you for your suggestion; while this removes the error page, the validation attributes are still not applying. I've put a video below where I both try the component as amended and without the additional validation error in the login function. Both of which garner similar results 

https://www.youtube.com/watch?v=DPkP9xkUo2I

**Reply by DanielSpravtsev** [reply]
Call $this->validate() before attempting to login. 

It you follow the docs and waiting for automatic validation - this is not working or maybe working only with .live properties.  😄  I don't know, here is my discussion about it https://github.com/livewire/livewire/discussions/6157 


And you can do not pass onUpdate: true to Rule, because it's true by default

**Reply by WheelieDev** [reply]
Having refactored to your suggestion, this is what happens now. 

<img width="1206" alt="Screenshot 2023-08-03 at 10 00 18" src="https://github.com/livewire/livewire/assets/5488110/2de9293a-221f-4c1e-b7e4-f55954ddf239">

I agree with your discussion, and it worked during the Laracon talk, so maybe a regression?

**Reply by WheelieDev** [reply]
I have managed to have a further tinker and you're idea kinda works if you remove the form and put the code straight in the component. However it only validates against the top attribute

**Comment by joshhanley** [comment]
@TomEasterbrook I think there is an issue with the auto-validation. Any chance you could submit a failing test PR? Thanks!

**Reply by WheelieDev** [reply]
Definitely up for giving it a go would be my first contribution to open source so any comments on the eventual PR would be very welcome.
